posthog/async_migrations/setup.py

```diff
@@ -33,18 +38,51 @@ def reload_migration_definitions():
 reload_migration_definitions()
 
 
+def is_fresh_install(first_migration: str):
+    # To verify we're on a fresh install
+    # (1) check that the first migration is not started
+    # (2) check that there are no events in the events table (i.e. someone didn't try to upgrade to 1.40 from before async migrations introduction)
+    migration_instance = AsyncMigration.objects.filter(name=first_migration).first()
+    if migration_instance.status != MigrationStatus.NotStarted:
+        return False
+    event_cnt = sync_execute("SELECT count() FROM events")[0][0]
+    if event_cnt > 0:
+        return False
+    return True
+
+
+def mark_all_existing_migrations_as_completed():
+    logger.info("Fresh install - marking all existing migrations as completed")
+    # we can tell by the percentage being 0 and time being the same that they were completed at startup
+    finished_at = now()
+    for migration_name in ALL_ASYNC_MIGRATIONS.keys():
+        sm = AsyncMigration.objects.filter(name=migration_name).first()
+        sm.status = MigrationStatus.CompletedSuccessfully
+        sm.finished_at = finished_at
+        sm.save()
+
+
+def check_required_migrations_have_been_completed(ignore_posthog_version: bool, applied_migrations):
```

Comment:
Missing typing for applied_migrations.

Why do we need to pass applied_migrations in the first place?